src/main/java/com/redhat/rhjmc/containerjfr/platform/ServiceRef.java

```diff
@@ -41,36 +41,37 @@
  */
 package com.redhat.rhjmc.containerjfr.platform;
 
+import javax.management.remote.JMXServiceURL;
+
 import org.apache.commons.lang.builder.EqualsBuilder;
 import org.apache.commons.lang.builder.HashCodeBuilder;
 import org.apache.commons.lang3.builder.ToStringBuilder;
 
 public class ServiceRef {
 
-    private final String connectUrl;
+    private final JMXServiceURL JMXServiceURL;
     private final String alias;
-    private final int port;
 
-    public ServiceRef(String connectUrl, int port) {
-        this(connectUrl, connectUrl, port);
+    public ServiceRef(JMXServiceURL JMXServiceURL) {
+        this(JMXServiceURL, JMXServiceURL.toString());
```

Comment:
It doesn't seem very useful to use the URL itself as the alias (which is intended to be a human-friendly identifier) for the URL.

Reply:
Should be fixed, haven't tested yet if the Optional gets serialized properly, I will do that next.